core/shared/platform/common/posix/posix_socket.c

```diff
@@ -351,6 +353,306 @@ os_socket_convert_sockaddr(struct sockaddr *addr, uint8_t *buf, size_t buflen,
     return BHT_OK;
 }
 
+int
+os_socket_setbooloption(bh_socket_t socket, int level, int optname,
+                        uint8 enable)
+{
+    int option = (int)enable;
+    if (setsockopt(socket, level, optname, &option, sizeof(option)) != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_getbooloption(bh_socket_t socket, int level, int optname,
+                        uint8 *enabled)
+{
+    assert(enabled);
+
+    int enabled_size = sizeof(enabled);
+    if (getsockopt(socket, level, optname, enabled, &enabled_size) != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_set_send_buf_size(bh_socket_t socket, size_t bufsiz)
+{
+    if (setsockopt(socket, SOL_SOCKET, SO_SNDBUF, &bufsiz, sizeof(bufsiz))
+        != 0) {
+        return BHT_ERROR;
+    }
+
+    return BHT_OK;
+}
+
+int
+os_socket_get_send_buf_size(bh_socket_t socket, size_t *bufsiz)
+{
+    assert(bufsiz);
+
+    socklen_t bufsiz_len = sizeof(bufsiz);
+    if (getsockopt(socket, SOL_SOCKET, SO_SNDBUF, bufsiz, &bufsiz_len) != 0) {
```

Comment:
I'd recommend using exact types defined for each property ([see here](https://pubs.opengroup.org/onlinepubs/009695399/functions/getsockopt.html)). For example, for `SO_SNDBUF` the value should be `int`.